presto-main/src/main/java/com/facebook/presto/metadata/DiscoveryNodeManager.java

```diff
@@ -301,6 +335,18 @@ private synchronized void refreshNodesInternal()
         // nodes by connector id changes anytime a node adds or removes a connector (note: this is not part of the listener system)
         activeNodesByConnectorId = byConnectorIdBuilder.build();
 
+        SetView<String> deadNodeIds = difference(nodes.keySet(), activeNodesBuilder.build().stream().map(InternalNode::getNodeIdentifier).collect(Collectors.toSet()));
```

Comment:
`nodes` contains all the active nodes we ever saw. So yes, if an active node goes inactive, then we consider it as a dead, which is fine i think because we don't want to schedule splits to it.